source/clear-linux/tutorials/kubernetes/kubernetes-bp.rst

```diff
@@ -0,0 +1,104 @@
+.. _kubernetes-bp:
+
+Kubernetes Best Practices on |CL|
+#################################
+
+This tutorial shows you how to manage your Kubernetes cluster while using
+:command:`swupd` to update |CL-ATTR|.
+
+In our tutorial :ref:`kubernetes`, we explain how to set up a Kubernetes
+cluster on |CL| using `kubeadm`. `Kubeadm documentation`_ often builds on the
+assumption that the distribution uses a traditional package manager (e.g.,
+RPM/DEB). In contrast, |CL| manages packages in bundles. This document
+describes best practices to manage cluster upgrades with kubeadm on a |CL|-based cluster.
+
+Prerequisites
+**************
+
+Let's review some basic assumptions:
+
+* |CL| includes Kubernetes support with the `cloud-native-basic` bundle.
+
+* Execute :command:`sudo swupd update` **on each node** to do the update.
+
+* Repeat the above command for both master and worker nodes.
+
+Executing this command updates all of the kubernetes node and client
+binaries simultaneously, which are part of `cloud-native-basic` bundle
+(including kubectl, kubeadm, kubelet).
+
+.. note::
+
+   Oher Linux\* distros shown in Kubernetes upgrade documentation reflect
+   `apt-get update`, `apt-mark hold kubeadm`, and similar commands; however, such commands **aren not valid** on |CL|.
+
+
+Update the cluster
+******************
+#. Read kubernetes documentation `before you begin`_.
+
+#. On your master node, run the command:
+
+   .. code-block:: bash
+
+      sudo swupd update
+
+#. Run this command to check the message-of-the-day, or `motd`:
+
+   .. code-block:: bash
+
+      cat /run/motd
+
+#. Next, ....[Explain resultant actions/choices from above]
+
```

Comment:
@mythi, Here we need to describe the options a user has. If motd?; if no motd?

Reply:
the motd info is available only if the kubernetes minor version changes (e.g., 1.12->1.13). there's nothing else than "please note". now that I think of if, it might be user friendly to have the message *always* when kubernetes version changes so that the users knows when to update.

Reply:
I agree that it would be best to always show the motd when the version changes. This can act as a trigger. Let's discuss in our meeting.